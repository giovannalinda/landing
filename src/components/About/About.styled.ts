import { motion } from 'framer-motion'
import styled from 'styled-components'
import { theme } from '~/styles'

import { Container as Button } from '../Button/Button.styled'

export const Container = styled.section`
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-gap: 5.6rem;
  align-items: center;
  padding-top: 13.2rem;

  @media screen and (max-width: ${theme.breakpoints.smallDesktop}) {
    justify-content: center;
    display: flex;
  }
`

export const AboutImageWrapper = styled(motion.div)`
  @media screen and (max-width: ${theme.breakpoints.smallDesktop}) {
    display: none;
  }
`

export const Content = styled(motion.div)`
  display: flex;
  text-align: left;
  flex-direction: column;

  h1 {
    max-width: 67.2rem;
    font-size: ${theme.font.sizes.extraTitle};
    font-weight: ${theme.font.weigths.medium};
    line-height: ${theme.font.lineHeights.extraTitle};
  }

  p {
    font-size: ${theme.font.sizes.large};
    line-height: ${theme.font.lineHeights.large};
    color: ${theme.colors.gray500};
    max-width: 67.2rem;
    padding-top: 2rem;
    padding-bottom: 3.6rem;
  }

  @media screen and (max-width: ${theme.breakpoints.mobile}) {
    text-align: center;

    h1 {
      font-size: ${theme.font.sizes.title};
      line-height: 5.2rem;
    }

    p {
      font-size: ${theme.font.sizes.small};
    }
  }
`

export const Carrer = styled.div`
  display: flex;
  text-align: left;
  gap: ${theme.space[8]};

  @media screen and (max-width: ${theme.breakpoints.mobile}) {
    gap: ${theme.space[4]};
    flex-direction: column;
  }
`

export const CarrerItem = styled.div`
  display: flex;
  align-items: center;

  h3 {
    color: ${theme.colors.secondaryPink};
    font-size: 5.6rem;
    line-height: 6.4rem;
    font-weight: ${theme.font.weigths.regular};
  }

  span {
    max-width: 9.5rem;
    margin-left: ${theme.space[4]};
    font-size: ${theme.font.sizes.large};
    line-height: ${theme.font.lineHeights.large};
  }

  @media screen and (max-width: ${theme.breakpoints.mobile}) {
    background: rgba(255, 255, 255, 0.015);
    border-radius: ${theme.borderRadius.lg};
    padding: ${theme.space[3]} ${theme.space[6]};

    h3 {
      width: 6rem;
      font-size: ${theme.font.sizes.subtitle};
    }

    span {
      font-size: ${theme.font.sizes.small};
      max-width: 100%;
    }
  }
`

export const SayHelloButton = styled(Button)`
  display: flex;
  align-items: center;
  margin-top: ${theme.space[6]};
  margin-right: auto;

  @media screen and (max-width: ${theme.breakpoints.mobile}) {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
  }

  svg {
    margin-left: ${theme.space[2]};
  }
`
