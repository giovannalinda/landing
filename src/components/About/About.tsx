import { CgArrowLongRight } from 'react-icons/cg'
import Image from 'next/image'

import { Assets } from '~/assets'

import * as S from './About.styled'

export function About() {
  return (
    <S.Container id='#about'>
      <Image src={Assets.About} width='592' height='574' />
      <S.Content>
        <h1>
          Software Engineer and Infrastructure specialist based in Brazil.
        </h1>
        <p>
          Morbi quam velit, euismod in imperdiet vitae, elementum et elit. Nunc
          finibus, felis sit amet sollicitudin sollicitudin, nisi magna feugiat
          enim, in maximus urna enim ac tortor. Nunc in volutpat ipsum, molestie
          commodo odio. Quisque auctor nisi mi. Aenean venenatis sapien et
          interdum interdum.
        </p>

        <S.Carrer>
          <S.CarrerItem>
            <h3>17+</h3>
            <span>Years of Experience</span>
          </S.CarrerItem>
          <S.CarrerItem>
            <h3>325+</h3>
            <span>Completed Projects</span>
          </S.CarrerItem>
        </S.Carrer>
        <S.SayHelloButton variant='purple' size='md' rounded={false}>
          SAY HI <CgArrowLongRight size={24} />
        </S.SayHelloButton>
      </S.Content>
    </S.Container>
  )
}
