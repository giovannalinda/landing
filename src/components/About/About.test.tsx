import { render } from '@testing-library/react'
import { About } from './About'

describe('Apresentation', () => {
  it('should match snapshot', () => {
    const { container } = render(<About />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
