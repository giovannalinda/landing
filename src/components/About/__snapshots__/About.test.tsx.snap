// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Apresentation should match snapshot 1`] = `
<section
  class="sc-gsnTZi brbedO"
  id="about"
/>
`;
